finder-app/writer.c

```diff
@@ -0,0 +1,136 @@
+/*
+* Code for AESD Assignment 2 for writer.sh script functionality in C language
+* @shrikant nimhan - shni9045
+*
+*/
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <syslog.h>
+#include <errno.h>
+
+int main(int argc,char **argv){
+
+    // Setup syslog with LOG_USER facility
+    openlog("Writer : ",LOG_PID,LOG_USER);
+
+    // Variables to check return value of file operations
+    int fd;
+    int nr;
+
+    char *filepath;                    // Stores user given file path
+    char *str_data;                    // Stores user given string to write
+
+    char *path;                        // Stores directory path 
+
+
+    // Temporary variables to operate on file path strings
+    char *temp_str;
+    char *temp;
+
+    char *command;
+
+    // Stat type strcuture to check for existence of directory through stat() call
+    struct stat path_stat={.st_mode=0};
+
+    // Error check for incorrect number of  paramters specified by user
+    if (argc < 3){
+
+        syslog(LOG_ERR,"ERROR:Few Arguments Specified\n");
+        return 1;
+
+    }
+
+    if (argc > 3){
+
+        syslog(LOG_ERR,"ERROR:Extra Arguments Specified\n");
+        return 1;
+
+    }
+
+    // Allocate memory to store path & string with null characters
+    filepath = (char*)malloc(sizeof(char)*(strlen(argv[1])+1));
+    str_data = (char*)malloc(sizeof(char)*(strlen(argv[2])+1));
```

Comment:
@shni9045 you oculd simplify logic here to use const char * as long as you write the code suchthat you don't need to modify these values.